src/components/GoodsList.tsx

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import './GoodsList.scss';
+
+interface Props {
+  goods: string[];
+}
+
+interface State extends Props {
+  reverse: boolean;
+  sortBy: string;
+  selectValue: string;
+}
+
+export class GoodsList extends React.Component<Props, State> {
+  state = {
+    goods: this.props.goods,
+    reverse: false,
+    sortBy: '',
+    selectValue: '1',
+  };
+
+  sortAlphabetical = () => {
+    this.setState({
+      sortBy: 'alphabetical',
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+  };
+
+  reverseGoods = () => {
+    this.setState(state => {
+      return ({
+        reverse: !state.reverse,
+      });
+    });
+  };
+
+  reset = () => {
+    this.setState({
+      reverse: false,
+      sortBy: '',
+      selectValue: '1',
+    });
+  };
+
+  selectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    this.setState({
+      selectValue: value,
+    });
+  };
+
+  render() {
+    const { reverse, sortBy, selectValue } = this.state;
+    const visibleGoods = this.state.goods.filter(good => good.length >= +selectValue);
+
+    if (sortBy) {
+      visibleGoods.sort((good1, good2) => {
+        if (sortBy === 'alphabetical') {
+          return good1.localeCompare(good2);
+        }
+
+        return good1.length - good2.length;
+      });
+    }
+
+    if (reverse) {
+      visibleGoods.reverse();
+    }
```

Comment:
make a separate function for this logic, do not write in between render and return